manifests/fcontext.pp

```diff
@@ -63,50 +63,42 @@
 # }
 #
 define selinux::fcontext (
-  $destination = '',
+  $pathname,
+  $destination = undef,
   $context     = '',
-  $pathname    = '',
   $filetype    = false,
-  $filemode    = '',
+  $filemode    = undef,
   $equals      = false,
-  $policy      = 'targeted',
 ) {
-  Exec {
-    path => '/bin:/sbin:/usr/bin:/usr/sbin',
-  }
-  if $pathname == '' {
-    fail('pathname must not be empty')
-  }
 
-  if ( $equals == false ) and ( $pathname == '' ) {
-    fail('pathname must not be empty')
+  if !defined(Class['selinux']) {
+    fail('You must include the selinux base class before using any selinux defined resources')
   }
 
-  if ( $equals == true ) and ( $destination == '' ) {
-    fail('destination must be set if equals is true')
+  validate_absolute_path($pathname)
+  validate_bool($filetype, $equals)
+
+  if $equals {
+    validate_absolute_path($destination)
   }
 
-  if ( $filetype == true ) and ( $filemode == '' ) {
-    fail('file mode must not be empty')
+  if $equals and $filetype {
+    fail('Resource cannot contain both "equals" and "filetype" options')
```

Comment:
Not so much in this module, but others I'm working on (mainly `puppet-nginx`). Also admit this might be a bit of scope creep, so maybe it's better to :football: on this one and re-address after this lands.